### PR TITLE
feat(replays): Add commit throttling to replays recording-segment consumer

### PR DESCRIPTION
**Pull request by cmanallen**
Prior art: https://github.com/getsentry/snuba/blob/master/snuba/subscriptions/executor_consumer.py#L431-L446

### CODE REVIEW

**Review by JoshFerge — COMMENTED** [review]
mind linking to the "prior art" if any, in the description?

File: src/sentry/replays/consumers/recording/process_recording.py [hunk 1]
```diff
@@ -173,48 +174,65 @@ def submit(self, message: Message[KafkaPayload]) -> None:
...
+                logger.warning(f"Timed out with {len(self.__futures)} futures in queue")
+                break
+
+            # Pop the future from the queue.  If it succeeds great but if not it will be discarded
+            # on the next loop iteration without commit.  An error will be logged.
+            message, future = self.__futures.popleft()
+
+            try:
+                future.result(remaining)
+                self.__commit({message.partition: Position(message.offset, message.timestamp)})
```
Comment (JoshFerge) [inline src/sentry/replays/consumers/recording/process_recording.py:203]: would we want to commit regardless of an error here? not sure if it matters too much.

**Review by JoshFerge — APPROVED** [review]
looks good. just the one small comment

File: src/sentry/replays/consumers/recording/process_recording.py (hunk 1 above)
Comment (cmanallen) [reply src/sentry/replays/consumers/recording/process_recording.py:203]: The way snuba handles it is to not commit and then die (leaving the queue full of items).  Not sure whats best to be honest.

Comment (JoshFerge) [reply src/sentry/replays/consumers/recording/process_recording.py:203]: cc @lynnagara

Comment (cmanallen) [reply src/sentry/replays/consumers/recording/process_recording.py:203]: Talked with lyn.  This commit behavior is fine.  We don't want to commit failed messages.  Should be able to retry.

Comment (JoshFerge) [reply src/sentry/replays/consumers/recording/process_recording.py:205]: sounds good. 🚢

**Review by lynnagara — APPROVED** [review]

**Pull request merged**